packages/dds/tree/src/shared-tree/schematizeTree.ts

```diff
@@ -112,9 +106,11 @@ export enum UpdateType {
 	Incompatible,
 }
 
+/**
+ * Returns how compatible updating checkout's schema is with the viewSchema.
```

Comment:
Since the update can only be either compatible or not now, how about we say something like:

```suggestion
 * Returns if updating checkout's schema is compatible with the viewSchema.
```

It does sound awkward considering this function doesn't return a boolean though. Is there a reason for us to keep the enum when there are only two possible values now?

Reply:
UpdateType has 3 possible values (down from 4). Github's diff view is just hiding one.

Even if there were just two, using a enum in such cases is often better for several reasons:

- You get explicit names for each state, making it more clear which is which and what exact details they are referring to.
- You can have doc comments on both the values and the enum itself to add additional clarity, and these docs follow the values through the code. If you just use an boolean and then pass it somewhere else or stave it in a variable, it becomes harder to document or refer to the documentation
- The extra type safety of the enum prevents a lot of possible errors, like using some other boolean where this on is expected.
- Any code taking in the enum makes it clear which things produce it so you don't get code making up its own values when it should be computed using a specific utility
- Using an enum makes changes adding more option in the future simpler

Reply:
Ah I see, that makes sense